src/actions/NamespaceAction.ts

```diff
@@ -7,7 +7,8 @@ export enum NamespaceActionKeys {
   NAMESPACE_REQUEST_STARTED = 'NAMESPACE_REQUEST_STARTED',
   NAMESPACE_SUCCESS = 'NAMESPACE_SUCCESS',
   NAMESPACE_FAILED = 'NAMESPACE_FAILED',
-  SET_ACTIVE_NAMESPACE = 'SET_ACTIVE_NAMESPACE',
+  TOGGLE_ACTIVE_NAMESPACE = 'TOGGLE_ACTIVE_NAMESPACE',
```

Comment:
Why do we need this?  Do you want to react to every click?  I'm not sure but perhaps it is better to process changes as a group, and set all namespaces at the same time.  This would prevent firing off requests that will quickly become obsolete.

Reply:
Reacting to every click is far easier and makes the UI look better (the url updates as i click), but I could set an strategy to wait for some time before actually changing the state.

Reply:
OK, I'll defer to your preference.  Don't worry about the wait, I guess.  I just didn't want to see the graph getting updated and then immediately re-updating and then again, etc.  But if it seems to work well for you then OK.